backend/fcm.py

```diff
@@ -0,0 +1,165 @@
+    """
+    if tokens:
+        result = push_service.notify_multiple_devices(
+            registration_ids=tokens, message_title=title,
+            message_body=body, message_icon=ICON_URL
+        )
+        return result
+
+
+def get_single_user_tokens(user_key):
+    """Calls get_tokens_from_firebase() 
+    to get all the tokens and then filter them.
+
+    Args:
+        user_key: The user ndb key.urlsafe().
+    
+    Returns:
+        The user's tokens or an empty list when the user hasn't
+        enabled notifications yet.
+    """
+    data = get_tokens_from_firebase(user_key)
+    tokens = filter_single_user_tokens(data)
+    return tokens
+
+
+def get_multiple_user_tokens(users_keys):
+    """This function calls get_all_tokens_from_firebase() 
+    to get all the tokens and then filter them using
+    filter_multiple_user_tokens function.
+
+    Args:
+        users_keys: The users ndb key.urlsafe().
+    
+    Returns:
+        The users' token or an empty list when None 
+        of the users haven't enabled notifications yet.
+    """
+    data = get_all_tokens_from_firebase()
+    tokens = filter_multiple_user_tokens(data, users_keys)
+    return tokens
+
+
+def get_all_tokens_from_firebase():
+    """This function only wraps the logic of 
+    make a request to firebase to retrieve all 
+    the tokens from the database.
+
+    Returns:
+        The request's content parsed to json.
+    """
+    response, content = _get_http().request(FIREBASE_TOKENS_ENDPOINT, method='GET')
+    return json.loads(content)
+
+
+def get_tokens_from_firebase(user_key):
+    """It gets all tokens from the firebase 
+    of the user whose key is user_key received as parameter.
+
+    Args:
+        user_key: The user's urlsafe key.
+    
+    Returns:
+        The request's content parsed to json.
+    """
+    firebase_endpoint = "%s/pushNotifications/%s.json" %(FIREBASE_URL, user_key)
+    response, content = _get_http().request(firebase_endpoint, method='GET')
+    return json.loads(content)
+
+
+def filter_single_user_tokens(content):
+    """It loops through the content keys
+    and for each object it appends the 
+    token property to the token's list.
+
+    Args:
+        content: A json returned from firebase.
+    
+    Returns:
+        The user's tokens.
+    """
+    tokens = []
+    for key in content.keys():
+        tokens.append(content[key]['token'])
+    return tokens
+
+
+def filter_multiple_user_tokens(content, users_keys):
+    """For each user, represented by user_key
+    It get the user's firebase objects and loops through
+    each object getting the token.
+
+    Args:
+        content: The json returned from firebase
+        users_keys: The users' keys who will receive the
+        notification. 
+    
+    Returns:
+        The users' tokens.
+    """
+    tokens = []
+    for user_key in users_keys:
+        if user_key in content:
+            current_firebase_objects = content[user_key]
```

Comment:
Is it possible to use the function above `filter_single_user_token` for each user?